### DISCUSSION TITLE
flink cdc 3.0.1 怎么解决bigint unsigned自动转为decimal20的问题

### DISCUSSION THREAD

**Discussion by yuyii** [apache/flink-cdc#3288] [Q&A] [answered]
使用flink cdc 3.0.1，mysql中表的主键是bigint unsigned，会被自动转为decimal。但sink端是starrocks，不支持decimal作为主键的，会报错。希望能用bigint的，这个问题如何解决呢
另外tinyint(1)也会出错，会被转为bool型，这个问题在cdc 3.0中也没找到解决办法，都是类型映射的问题。
在此请教。

**Comment by yuyii** [comment]
bigint的问题我看到在3.1中会解决 https://issues.apache.org/jira/browse/FLINK-34690
tinyint的问题怎么处理呢

**Reply by jeeinn** [reply]
遇到了 tinyint(1) 转 boolean 的问题，楼主后来如何解决的。配置项确实不起作用

**Comment by yuxiqian** [comment]
Hi @yuyii, thank you for reporting this. Could you please clarify your problem? Are you trying to use `TINYINT(1)` as (one of?) primary key columns in StarRocks? I noticed in [StarRocks docs](https://docs.starrocks.io/docs/table_design/table_types/primary_key_table/) `BOOL` and `TINYINT` are both supported as primary keys:

> The primary key columns support the following data types: numeric (including integers and BOOLEAN), string, and date (DATE and DATETIME).

**Reply by yuyii** [reply]
The tinyint is not about primary key problem. I means while some field with tinyint(1) type in source table will be tranfered in bool type to  starrocks. But it's wrong. 
I try to add flink cdc source property as below:

jdbc.properties.tinyInt1isBit: "false"
jdbc.properties.transformedBitIsBoolean: "false"

But they have no effect.

**Reply by yuxiqian** [reply, marked as answer]
Hi @yuyii, the JDBC properties should not work since MySQL CDC captures changing data by parsing BinLog instead of establishing JDBC connection. I could reproduced this problem with the following setup:

```yaml
# MySQL source schema:
CREATE TABLE fallen_angel(
     ID VARCHAR(177) NOT NULL,
     BOOLEAN_COL BOOLEAN, 
     TINYINT_COL TINYINT(1),
     PRIMARY KEY (ID)
);

# StarRocks auto-created schema:
Field           Type            Null    Key     Default Extra
ID              varchar(531)    NO      true    NULL       
BOOLEAN_COL     boolean         YES     false   NULL    
TINYINT_COL     boolean         YES     false   NULL 
```

I believe it's a CDC type mapping issue, and created a JIRA ticket [FLINK-35308](https://issues.apache.org/jira/browse/FLINK-35308) to track this problem. Thanks for reporting this!
